app/src/main/java/com/x8bit/bitwarden/data/auth/repository/AuthRepositoryImpl.kt

```diff
@@ -1256,6 +1258,32 @@ class AuthRepositoryImpl(
                 },
             )
 
+    override suspend fun verifyEmailToken(
+        email: String,
+        emailVerificationToken: String,
+    ): VerifyEmailTokenResult =
+        identityService
+            .verifyEmailToken(
+                VerifyEmailTokenRequestJson(
+                    email = email,
+                    emailVerificationToken = emailVerificationToken,
+                ),
+            )
+            .fold(
+                onSuccess = {
+                    VerifyEmailTokenResult.Verified
+                },
+                onFailure = {
+                    // Server sends a message if the link is expired, this is the only way
+                    // to check if the token is expired
+                    if (it.message?.contains("Expired link", ignoreCase = true) == true) {
```

Comment:
I just spoke to @dseverns-livefront about this. Stuff like this should be in the service.